code/player/TTTPlayer.Lookup.cs

```diff
@@ -0,0 +1,50 @@
+using Sandbox;
+
+namespace TTTReborn.Player
+{
+    partial class TTTPlayer
+    {
+        public TTTPlayer Player { get; set; }
+
+        private const float MAX_DRAW_DISTANCE = 500;
+
+        private void TickAttemptLookup()
+        {
+            using (Prediction.Off())
+            {
+                To client = To.Single(this);
+                TTTPlayer player = IsLookingAtPlayer();
+
+                if (player != null)
+                {
+                    // Send the request to the player looking at the player corpse.
+                    // https://wiki.facepunch.com/sbox/RPCs#targetingplayers
+
+                    if (!player.IsValid())
+                    {
+                        return;
+                    }
+
+                    SetNameplateHealth(client, player.Health);
```

Comment:
If we would use a struct like suggested, we don't need to send the full exact health but could instead just send a string as a health indicator (used to fetch the right state). But as AC doesn't care as much, we don't need this.